pymc_marketing/mmm/models/components/lagging.py

```diff
@@ -0,0 +1,313 @@
+                name="channel_adstock",
+                var=geometric_adstock(
+                    x=data,
+                    alpha=adstock_alpha,
+                    l_max=self.max_lagging,
+                    normalize=True,
+                    axis=0,
+                ),
+                dims=("date", "channel"),
+            )
+
+
+class WeibullPDFAdstockComponent(BaseFunction):
+    """
+    A class representing a Weibull PDF adstock component in a marketing mix model.
+
+    Parameters:
+    -----------
+    max_lagging : int
+        The maximum lagging value for the adstock component.
+    model : pymc.Model
+        The PyMC model object.
+    model_config : dict
+        The configuration dictionary for the model.
+
+    Attributes:
+    -----------
+    REQUIRED_KEYS : list
+        The list of required keys in the model configuration.
+
+    Methods:
+    --------
+    apply(data: Union[pm.Data, pm.MutableData]) -> pm.Deterministic:
+        Apply the Weibull PDF adstock component to the given data.
+
+    """
+
+    REQUIRED_KEYS = ["adstock_lambda", "adstock_shape"]
+
+    def __init__(self, max_lagging, model, model_config):
+        super().__init__(max_lagging, model)
+        self.model_config = model_config
+        _validate_model_config(
+            required_keys=self.REQUIRED_KEYS, model_config=self.model_config
+        )
+
+    def apply(self, data: Union[pm.Data, pm.MutableData]) -> pm.Deterministic:
+        """
+        Apply the Weibull PDF adstock component to the given data.
+
+        Parameters:
+        -----------
+        data : Union[pm.Data, pm.MutableData]
+            The data to apply the adstock component to.
+
+        Returns:
+        --------
+        pm.Deterministic
+            The adstocked data as a PyMC Deterministic variable.
+
+        """
+        self.adstock_lambda_dist = _get_distribution(
+            dist=self.model_config["adstock_lambda"]
+        )
+
+        self.adstock_shape_dist = _get_distribution(
+            dist=self.model_config["adstock_shape"]
+        )
+
+        with pm.modelcontext(self.model):
+            adstock_lambda = self.adstock_lambda_dist(
+                name="adstock_lambda",
+                dims="channel",
+                **self.model_config["adstock_lambda"]["kwargs"],
+            )
+
+            adstock_shape = self.adstock_shape_dist(
+                name="adstock_shape",
+                dims="channel",
+                **self.model_config["adstock_shape"]["kwargs"],
+            )
+
+            return pm.Deterministic(
+                name="channel_adstock",
+                var=weibull_adstock(
+                    x=data,
+                    lam=adstock_lambda,
+                    k=adstock_shape,
+                    type="PDF",
+                    l_max=self.max_lagging,
+                ),
+                dims=("date", "channel"),
+            )
+
+
+class WeibullCDFAdstockComponent(BaseFunction):
```

Comment:
Personally think that both of the weibull's can be combined. 

Another reason to just pass the class itself 
For instance, 
```python
adstock_cdf = WeibullAdstock(max_lag=12, kind="CDF")
adstock_pdf = WeibullAdstock(max_lag=12, kind="PDF")

# Choose the adstock out of the class 
adstock = adstock_cdf
mmm = DelayedSaturatedMMM(
    ..., 
    adstock=adstock, 
    ...,

Reply:
Agree this can be simplified. Let me think about it a bit more, I'm not fully convinced with the classes because decrease our control and users will need to import the class to use it, none strong opinion here but I recognize doing it has a few things in favor.

Reply:
Our decrease in control = more control for users. 

If the fear is having to import one more thing, then I think their could be a route to use strings or classes. However, having a non-editable mapping from strings to classes shouldn't be hard coded and non-flexible to user. imo